### ISSUE TITLE
[Request] Conditions that actually auto-track.

### ISSUE THREAD

**Issue by NuptupTDOW** [EndlesNights/dnd4eBeta#243]
Is there a plan to add/change the conditions that can be applied to tokens to match the conditions present in DnD 4e and have the applicable math updates accurately reflect on the character's stats and defenses?

Some key examples that my group encounters a LOT would be;

Granting C.A.: Just generic C.A. from a power or whatever that causes all defenses to go down by 2. This one should be separate from. . .

Flanking C.A.: which shares the same penalty type as the above, but would be somewhat automated for detection with player-owned tokens detecting if they're flanking. Alternatively, make it a condition the player can apply to themselves such as. . .

Has C.A.: Which gives the player a flat +2 to attacks, but have this type be checked against the generic "Granting C.A." from above so that the character doesn't double dip from C.A. This "Has C.A." could be made much easier to implement if request #225 is also implemented.

Prone: Automatically reduces speed on the character sheet to half rounded down (mainly for drag ruler to show the reduced speed, thus reminding people they're prone), also make it automatically give the Granting C.A. condition (though, this would need some work around vs. ranged, but I'm not sure how to do that effectively. 

For the most part, that same style of automation would be an EXCELLENT addition to the system, and I'd love to see that be a focus if possible. It would also be very convenient if these conditions were simply added to a compendium that would allow players and gm's to directly add these as effects to powers and items. For example, Sweeping Cut (Barbarian/Berserker Daily) causes all targets hit by the attack to fall prone. Being a Close Burst 1, it can hit up to 8 people, most often 3-4. So, it would be excellent if we could just have the effect "Prone" be applied on success and actually reflect that in the math. We currently have it apply a " "Prone" " effect with the same icon as the default Prone condition, but it would be better if it was able to directly add a pre-coded effect, and for those pre-coded effects to be conditions that are apply-able to the tokens normally as well.

Lastly, it would be great for marks to more easily track who applied the mark, and then apply the mark effects based on who the enemy targets. This could be done potentially by allowing the player to control certain conditions on enemies, or through the use of a power that applies a marked condition to the enemy token, both usable by the player doing the marking, and then having that mark check the enemy's target and apply the penalty so long as "Defender Character Name" is not a selected target.

Anyway, I'm rambling, but I feel like you probably get my point, so I'm hoping you can think it over and see what might be possible.

**Comment by FoxLee** [comment]
These may not be appropriate to auto-apply—or at least not as described. For example, implementing combat advantage as a penalty to defences is not a good idea, because you're inviting incorrect stacking with other CA sources. Just for an example, if the attacker uses the normal way to apply combat advantage implementation (the checkbox in the roll dialogue) they would effectively double-dip, completely by accident.

I agree that combat advantage is important given how ubiquitous it is. But I don't think we should be considering any implementation that don't integrate correctly with the one that's already there. It has so many conditional cases that you really need the "choose to apply on a per-roll basis". You already noted how prone's CA is conditional, and there's also a slew of effects that grant CA only on the next attack, or only to certain enemies or attack types, or even prevent things from gaining CA.

Now, if we could work out a condition that instead auto-checked the CA bonus—crucially meaning you still have the chance to turn it off when you roll, if it shouldn't apply to this particular attack—that would be ideal. (It would also support the possibility for future CA-sensitive automation, like damage bonuses that rely on CA.)

**Comment by NuptupTDOW** [comment]
That is a pretty good way to look at it, going with a condition automatically applying C.A. toggle to correct targets, which would help it with multi-target attacks and whatnot as well. But, beyond JUST C.A., there's other conditions that would certainly be nice to have automatically affect the target, such as slow applying the move speed penalty for example.

**Comment by NuptupTDOW** [comment]
I'd like to list out all of the "standard" conditions in 4e and see if I can list out all of the status effects that could/should be automatically applied to a token when they are subject to that condition. This is mostly just to get the topic rolling and serve as a staging ground for this as an idea. Address any concerns referencing C.A. to #215 as to not muddy up this topic. For the purposes of this request, simply assume that "Toggling C.A." will correctly apply it. This topic isn't about C.A., only about the conditions that can apply it and what those conditions would need to do. Likewise, this topic is not going to address Marks/Marking/Defender Auras at all, as that will be it's own entirely different issue that I will write up. 

1. Blinded
- Applies a -10 penalty to Perception Checks. (Unsure type, using the wiki at the moment)
- Toggles C.A.
- Prevents this Actor from generating C.A. during Flanking Calculations.
- Sets the token's "Has Vision" toggle to false.

2. Dazed
- Toggles C.A.
- Prevents this Actor from generating C.A. during Flanking Calculations.
- Limit action usage if action tracking is ever implemented. (Honestly, I don't think this is any priority at all)

3. Deafened
- Applies a -10 penalty to Perception Checks.
- Removes the token's ability to hear any environmental sounds. (I'm not sure if this is possible..?)

4. Dominated
- Toggles C.A.
- Prevents this Actor from generating C.A. during Flanking Calculations.

5. Dying
- Applies Unconscious.

6. Grabbed
- Applies Immobilized.

7. Helpless
- Toggles C.A.

8. Immobilized
- Sets token speed to 0.

9. Petrified
- Applies Unconscious.
- Applies 20 to all resistances.
- Applies a color shift to the token if that's possible.

10. Prone
- Applies a -2 penalty to all attack rolls.
- Applies a +2 to all defenses against attacks made from non-adjacent. (Not exactly sure on the wording since I'm working off of the wiki. Will update later.)
- Toggles C.A. for enemies that are adjacent to the token.

11. Removed From Play
- Toggles the token's visibility on the game board for the duration, effectively removing it from the player's control.

12. Restrained
- Applies Immobilized. (I think? Will fix later.)
- Applies a -2 penalty to all attack rolls.
- Toggles C.A.

13. Slowed
- Applies a penalty to the actor's speed to set it to 2 if it's speed was higher.

14. Stunned
- Toggles C.A.
- Prevents this Actor from generating C.A. during Flanking Calculations.
- Prevent action usage if action tracking is ever implemented. (Honestly, I don't think this is any priority at all)

15. Surprised
- Toggles C.A.
- Prevents this Actor from generating C.A. during Flanking Calculations.
- Prevent action usage if action tracking is ever implemented. (Honestly, I don't think this is any priority at all)

16. Unconscious
- Applies Blinded.
- Applies Deafened.
- Applies Helpless.
- Applies Prone.
- Applies a -5 penalty to all defenses.
- Prevents this Actor from generating C.A. during Flanking Calculations.
- Limit action usage if action tracking is ever implemented. (Honestly, I don't think this is any priority at all)

17. Weakened
- Reduces all damage generated from the actor's attacks by half. (Assuming we can have an isAttack property, similar to the isBasic added by #393 this would allow us to filter out automated damage such as flurry of blows, magic missile, and ongoing damage to properly respect which damage types should be reduced.)

Overall, I think most of these can be established as system-wide defined conditions with built-in functionality. If that's possible, then the Effects dialog for all kinds of things can directly apply THAT EXACT condition, instead of applying the effects of said condition. By applying an exact condition with a known value, we can then check for that value on tokens when applying other effects. 

For example, if we apply THE Slowed condition to a target, we could then have any character using the World Serpent's grasp feat automatically get their benefits when attacking that slowed target, such as causing their attack that hits a slowed target to automatically apply THE Prone condition. 

Another example would be for things like Headmans Chop. If THE Prone condition is assigned to a target, then a character with that feat could have a check in their weapon's damage line that checks to see if the target of the attack has THE Prone condition, and if they do, it adds the damage to the roll automatically.

I'm fairly sure this is all possible (setting up system-wide conditions, assigning them specific I.D.'s, and having those conditions apply set effects), with the exception currently of Flanking Detection, C.A. calculations, action tracking/limiting, and the distance-related conditional bonuses related to being Prone. I'm pretty certain that all other things listed above can be done through the effect options we already have available to us, and mostly would just need to be assigned to a system-wide condition instead of being manually coded every time you want that type of condition to be applied.

**Comment by FoxLee** [comment]
I don't have time to give this a really detailed answer right now, just wanted to quickly address one part; if you're using the statuses/conditions array in the effect config, it already _is_ applying the "actual" condition. That's why you see it toggled in the manual conditions list.

I see what you're saying—it would be good if conditions imposed mechanical effects in and of themselves—but what's preventing that is not that the system isn't applying "the" condition, it's that "the" condition has no mechanical meaning.

**Comment by NuptupTDOW** [comment]
> it's that "the" condition has no mechanical meaning.

Yeah, for the most part, that point right there was the entire purpose of my post. I guess I should have clarified that a bit more.

**Comment by FoxLee** [comment]
I think we've finished the last of this now!
